src/core/sorted_map.cc

```diff
@@ -63,6 +73,26 @@ double GetObjScore(const void* obj) {
   return absl::bit_cast<double>(absl::little_endian::Load64(ptr));
 }
 
+void SetObjScore(void* obj, double score) {
+  sds s = (sds)obj;
+  char* ptr = s + sdslen(s) + 1;
+  absl::little_endian::Store64(ptr, absl::bit_cast<uint64_t>(score));
+}
+
+// buf must be at least 10 chars long.
+void* BuilScoredKey(double score, bool is_str_inf, char buf[]) {
+  buf[0] = SDS_TYPE_5;  // length 0.
+  buf[1] = 0;
+  absl::little_endian::Store64(buf + 2, absl::bit_cast<uint64_t>(score));
+  void* key = buf + 1;
+
+  // to exclude the double score we set the secondary string to +inf.
+  if (is_str_inf) {
+    key = (void*)(uint64_t(key) | kInfTag);
+  }
+  return key;
+}
```

Comment:
Please add a comment that this is only for temporary non-stored entries for querying as I was double checking whether those tagged pointers can actually be stored in the tree itself

just a nit: I always prefer to use spans to assert the length, but its not important

Reply:
added a comment inside the operator, also added a comment about infinity semantics.